### ISSUE TITLE
LearningRateMonitor wont log if interval = 'step' or if scheduler name is not declared

### ISSUE THREAD

**Issue by fmellomascarenhas** [PyTorchLightning/pytorch-lightning#6951]
## 🐛 Bug

LearningRateMonitor wont log lr if a name is not declared or if interval = 'step'

works
```
 scheduler = {'scheduler': torch.optim.lr_scheduler.OneCycleLR(optimizer,
                                                            max_lr=[cfg.max_lr/cfg.encoder_lr_frac, cfg.max_lr],
                                                            epochs=self.epochs,
                                                            steps_per_epoch = len(self.loaders_dict['train'])),
                     'name': 'random_name',
                     'frequency': 1,
                     'interval': 'epoch,
                     "monitor": 'train_loss'}
```

doesnt work with step
```
 scheduler = {'scheduler': torch.optim.lr_scheduler.OneCycleLR(optimizer,
                                                            max_lr=[cfg.max_lr/cfg.encoder_lr_frac, cfg.max_lr],
                                                            epochs=self.epochs,
                                                            steps_per_epoch = len(self.loaders_dict['train'])),
                     'name': 'random_name',
                     'frequency': 1,
                     'interval': 'step', #CHANGED TO STEP
                     "monitor": 'train_loss'}
```

doesnt work without name
```
 scheduler = {'scheduler': torch.optim.lr_scheduler.OneCycleLR(optimizer,
                                                            max_lr=[cfg.max_lr/cfg.encoder_lr_frac, cfg.max_lr],
                                                            epochs=self.epochs,
                                                            steps_per_epoch = len(self.loaders_dict['train'])),
                    # 'name': 'random_name',
                     'frequency': 1,
                     'interval': 'epoch',
                     "monitor": 'train_loss'}
```
Used linux, pytorch 1.7, latest version of PL

## Source of error

In the callback, on_train_batch_start, the function _should_log finds that **trainer.log_every_n_steps = 50**.
If i set trainer.log_every_n_steps = 1, then the logger works. So I assume that when scheduler['interval'] = 'step', the trainer doesnt updated the log_every_n_steps.

```
def on_train_batch_start(self, trainer, *args, **kwargs):
        if not self._should_log(trainer):
            return

        #[logging code]

def _should_log(trainer) -> bool:
        should_log = ((trainer.global_step + 1) % trainer.log_every_n_steps == 0 or trainer.should_stop)

        return should_log
```

**Comment by justusschock** [comment]
Hi @fmellomascarenhas, 
You're right, we keep the logs internally, but just don't flush them to the logger every step.

Flushing is a very time consuming operation that would slow down training significantly, if done at every iteration. 

Do you think this should be changed? Since actually, there is a flag to change this (as you noted)

**Comment by fmellomascarenhas** [comment]
Hi @justusschock, thanks for answering! I think that there are two fixes:

1) the logging should have the same behavior, regardless of the scheduler name being provided;
2) if the developer explicitily declares in the scheduler "frequency: 1", then the frequency should be 1:

Do you agree?

**Comment by justusschock** [comment]
Hi @fmellomascarenhas 

For 1.) I agree!

For 2.) Actually I don't think so. Since the frequency of the scheduler and the frequency of the logger are two completely separated things. That means, that the lr is updated by the scheduler, just not yet logged (due to logger frequency).

Would you mind sending a PR for 1.)?
